nautobot/extras/models/datasources.py

```diff
@@ -170,3 +171,12 @@ def on_commit_callback():
             self.__initial_slug = self.slug
 
         transaction.on_commit(on_commit_callback)
+
+    def clean(self):
+        """Overload clean to enforce a repo with duplicate url does not have duplicate provided_contents."""
+        duplicate_repos = GitRepository.objects.filter(remote_url=self.remote_url).exclude(id=self.id)
+        if duplicate_repos.exists():
+            for content in self.provided_contents:
+                if duplicate_repos.filter(provided_contents__contains=content).exists():
+                    raise ValidationError(f"Duplicate GitRepository detected providing {content}")
```

Comment:
I feel like this error message should also include the URL. E.g. "Duplicate exists with this URL that provides this content"?

Reply:
You are creating the repo so you are already passing in the single attr for URL. The issue is less about duplicate URL but more about 2 to N GitRepositories providing the same content. You could absolutely have different GitRepository objects tracking different branches and each one provides different content. This is mainly to block the ability for a duplicate repo to provide `extras.job` because it will cause failures for `migrate` and `post_upgrade` until the duplicate repo is removed.

Reply:
Should the duplicate check only apply to repos that have the same specified `branch` as well as URL?

Reply:
I see that as a possible scenario BUT that would mean in the scenario I ran into today a user created 2 repos with different branches, one branch had all the original jobs plus an additional job so url + branch wouldn't have caught it.